helm-charts/splunk-otel-collector/templates/deployment-cluster-receiver.yaml

```diff
@@ -227,6 +271,18 @@ spec:
       - name: messages
         emptyDir: {}
       {{- end }}
+      {{- if (eq (include "splunk-otel-collector.persistentQueueEnabledMetrics" .) "true") }}
+      - name: persistent-queue-metrics
+        hostPath:
```

Comment:
I think statefulset would be better, right?
I'm not sure if we **strictly** need PVC.

Reply:
> I'm not sure if we strictly need PVC.
How would you get a persistent volume for the pod to persist its queue if not with a pvc? Switching to statefulset but still using hostPath will still mean the user needs to pin pod to a specific node. A persistent volume on the other hand means the queue moves with the pod when rescheduled.

Reply:
I see.
If we go ahead with PCV, we would need to use either `local` or `hostPath` configs. `hostPath` is out of the picture, as it will only work in single node env. 
Now, for `local` as well, we need to specify `nodeSelector`.
From https://kubernetes.io/docs/concepts/storage/volumes/#local,

> You must set a PersistentVolume nodeAffinity when using local volumes. The Kubernetes scheduler uses the PersistentVolume nodeAffinity to schedule these Pods to the correct node.

I guess this contradicts our approach. 

I think the other way to achieve this would be to use CSI for specific vendors based on `distribution` param.

Reply:
It will be left to the user to set the `storageClass` for the pvc. We don't need to get into the details of the type of storage used in our chart with pvc.

Reply:
@dmitryax I think you mentioned that this PR might be more of a parity thing for sck to otel, in which case should we keep the clusterReceiver out of this? If desired we can create a different issue to explore how best to add persistent queues for deployment (sts + pvc would be ideal)

Reply:
SGTM, let's exclude it for now and put a not in the values.yaml that the persistence is not applicable to the k8s cluster metrics and k8s events. The VPC's approach can be handled after

Reply:
Okay.